### DISCUSSION TITLE
Mark Print as Failed

### DISCUSSION THREAD

**Discussion by nickecs** [OctoFarm/OctoFarm#1154] [Feature Request]
Looking to add the ability to mark prints as failed at harvest and also edit completed print job status. Adding an additional button below print harvested would keep amount of clicks the same in a workflow versus a popup asking if the print was successful. 

![image](https://user-images.githubusercontent.com/104113470/176445754-f742d8ce-7602-4977-a7ef-352716131033.png)

**Comment by NotExpectedYet** [comment]
Will think about how to work this in. Currently there's no access to the historical record thats created from these functions that appear here. 

I'd rather not do some kind of match for the last file name printed but it's a good idea! 

The editing of the print job status can certainly be done though!

**Comment by nickecs** [comment]
I haven't dug into how it works so could be entirely off track, but my first though could be to store a job ID in the operations card buttons so that when Failed is selected, it runs an update query with the job ID and printer ID to change the status. 

Also credit to nftrix on discord as he mentioned it about a month ago but didn't see anything here.

**Reply by NotExpectedYet** [reply]
I'll credit you both if it gets imemented. It's better here as I forget and here is where I check most for issues and such. 

Yeah currently the history ID that's created isn't known until the print finished event triggers. That's a decent idea though as it would be much less work to generate the ID on the print start command and follow it through the job until complete. I'd just need to rework a few triggers and the history class to allow partially creating it.

**Comment by taker218** [comment]
I have just seen this discussion here and would like to add my vote to this.

I recently had some prints, that ran overnight and were finished the next morning, but they failed (part came loose and I had spaghetti all over the build plate) but in Octofarm the print is saved as successful, because it finished. Currently there seems to be no way to edit the status in the history, which would be super nice to have.

Either as a button when harvesting the print or as an editable field in the history tab.

**Comment by NotExpectedYet** [comment]
Just to let you know, I'm going to be looking at adding this over the next few months. 

History needs re-wroking to create the ID as the print is started then I'll be able to access the last prints ID in the front end. 

I will add this button and some other features. 

You can thank the patreons for this as they all voted for me to focus on Filament Manager and History as part of the next 3 months of development.
